### PR TITLE
Fix Issue #5 and plug some memory leaks

### LINKED ISSUE #5: Buffer overflows when provided parameter is larger than 26 characters
This issue is caused by the `path` variable defined in:

https://github.com/mt-empty/shortcut-c-client/blob/4aba42b663e1dd1c75c0d3d5971c7aa7ffd2c509/shortcut.c#L139

having a hardcoded size limit. Will have to expand `path`'s buffer based on the arguments length before calling strcat.

### PR DESCRIPTION
**Pull request by MadeOfMagicAndWires**
Fixes #5 by allocating enough space for the full path (up to BUFFER_SIZE + 24) and also frees some leftover allocated memory.

Ideally, I'd like for the PAGES_BASE_DIR etc. to be configurable, but that is for another day.

### CODE REVIEW

File: shortcut.c [hunk 1]
```diff
@@ -275,7 +288,7 @@ int getFileContent(char const *path, char **out)
     if (fseek(fp, 0, SEEK_SET))
         goto error;
 
-    *out = (char *)malloc(len);
+    *out = (char *)calloc(len+1, sizeof(char));
```
Comment (MadeOfMagicAndWires) [inline shortcut.c:296]: Using calloc because valgrind complains when calling strlen on uninitialized memory.

**Comment by mt-empty** [comment]
Looks good, thank you for your contribution.

> Ideally, I'd like for the PAGES_BASE_DIR etc. to be configurable, but that is for another day.

Same here.

**Review by mt-empty — APPROVED** [review]

**Comment by MadeOfMagicAndWires** [comment]
Fixed smaller than logic which I'm sure I had written down correctly at *some* point. 

I think that was the last of it though.

**Comment by MadeOfMagicAndWires** [comment]
Fixed another bug (in my brain); strncat != strncpy. 

That's really the last of it though, I'm going to get some breakfast.

(I also rebased the branch to keep the log clean)

**Review by mt-empty — APPROVED** [review]

**Pull request merged**